lib/topological_inventory/schema/default.rb

```diff
@@ -123,6 +124,80 @@ def add_tagging_collection(model, manager_ref: [:source_ref])
         end
       end
 
+      def add_service_instance_tasks
+        add_collection(:service_instance_tasks, inventory_collection_builder, {}, {:without_model_class => true}) do |builder|
+          add_default_properties(builder)
+          add_default_values(builder)
+
+          builder.add_dependency_attributes(
+            :service_instances => ->(persister) { [persister.collections[:service_instances]] }
+          )
+
+          save_block = lambda do |source, tasks_collection|
+            service_instance_tasks_custom_save(source, tasks_collection)
+          end
+
+          builder.add_properties(:custom_save_block => save_block)
+        end
+      end
+
+      def service_instance_tasks_custom_save(source, tasks_collection)
+        service_instance_collection = tasks_collection.dependency_attributes[:service_instances]&.first
+        src_refs = service_instance_collection.data.collect { |inventory_object| inventory_object.source_ref }
```

Comment:
`&.first` [here](https://github.com/RedHatInsights/topological_inventory-core/pull/188/files#diff-bf581c67a3bfa3b364180f998a222a83R145) implies this might be nil, but then you do `service_instance_collection.data.collect` which would fail

Reply:
Moved discussion over here https://github.com/RedHatInsights/topological_inventory-core/pull/187/files#r411515073